hypertrace-trace-enricher/hypertrace-trace-enricher-impl/src/main/java/org/hypertrace/traceenricher/enrichment/enrichers/resolver/backend/RedisBackendResolver.java

```diff
@@ -25,19 +26,21 @@ public RedisBackendResolver(FQNResolver fqnResolver) {
 
   @Override
   public Optional<Entity> resolveEntity(Event event, StructuredTraceGraph structuredTraceGraph) {
-    if (!SpanAttributeUtils.containsAttributeKey(event,
-        RawSpanConstants.getValue(Redis.REDIS_CONNECTION))) {
+    Optional<String> backendURI = DbAttributeUtils.getRedisURI(event);
+
+    if (backendURI.isEmpty()) {
       return Optional.empty();
     }
-    String backendURI =
-        SpanAttributeUtils.getStringAttribute(event, RawSpanConstants.getValue(Redis.REDIS_CONNECTION));
-    if (StringUtils.isEmpty(backendURI)) {
+
+    if (StringUtils.isEmpty(backendURI.get())) {
       LOGGER.warn("Unable to infer a redis backend from event: {}", event);
       return Optional.empty();
     }
-    final Builder entityBuilder = getBackendEntityBuilder(BackendType.REDIS, backendURI, event);
+
+    final Builder entityBuilder = getBackendEntityBuilder(BackendType.REDIS, backendURI.get(), event);
     setAttributeIfExist(event, entityBuilder, RawSpanConstants.getValue(Redis.REDIS_COMMAND));
     setAttributeIfExist(event, entityBuilder, RawSpanConstants.getValue(Redis.REDIS_ARGS));
+    setAttributeIfExist(event, entityBuilder, OTelDbAttributes.DB_CONNECTION_STRING.getValue());
```

Comment:
note : an entity should have its own set of attribute keys which we should set, this looks like a union of keys.

Reply:
No, this is a single constant.

Reply:
Do you think, we may have to derive  `REDIS_COMMAND ` and `ARGS` from `OTelDbAttributes.DB_CONNECTION_STRING`?

Reply:
Well, I am not sure if it is needed (since no consumer for it). I can add it later.